### ISSUE TITLE
Example of Formatting Setup

### ISSUE THREAD

**Issue by RapidOwl** [enhance-dev/enhance-starter-project#30]
I really wanted to try Enhance, but very quickly ran into auto-formatting issues. I can't get any of the following to automatically format template literals, especially in `mjs` files:
* ESLint
* Prettier
* VS Code itself

After installing `literally-html` I was getting an _interesting_ bug where every time it formatted my code, it malformed the template literal e.g.
```
return html`
	<header>
		<h1>Header</h1>
		<nav>
			<a href="/">home</a>
			<a href="/about">about</a>
		</nav>
	</header>
`;
```
was transformed into
```
return html`
	<header>
		<h1>Header</h1>
		<nav>
			<a href= />home</a>
			<a href=/about>about </a> </nav> </header>	</header>
`;
```

Would it be possible to provide an example of how to setup an Enhance project with auto-formatting? Maybe it could be added to the starter template? I'd be happy to make a PR for this, once I've got my own setup correct.

**Comment by kristoferjoseph** [comment]
I am very open to anything you feel would make the experience better. 

Formatting can be a contentious topic since developers tend to have their own preferences on editors and formatting so I was waiting to hear from users as to what they wanted. 

Personally I use eslint in vim so I’m not sure my personal setup would help much as I don’t rely on auto-formatting at all.

**Comment by macdonst** [comment]
@RapidOwl I hate to be that guy asking for a reproduction scenario but can you provide us with a bit more information so we can try to mirror your environment as closely as possible?

What are the versions of:

- ESLint
- Prettier
- VS Code itself
- Literally-html

Also, what operating system and version are you using? We often assume folks are on a Mac but you could be on Windows and I wonder if the above is a CRLF issue.

**Comment by RapidOwl** [comment]
Sorry, my bad. I should have provided more detail. Thank you so much for the quick replies though!

The following is my set up, but it's irrelevant now because of your last sentence. I changed my VS Code EOL setting for my Enhance project workspace to `\n (LF)` instead of `\r\n (CRLF)` and it now formats the template literals correctly!

I'm running:
* Windows 10
* VS Code 1.80.1
  * es6-string-html 2.12.1
  * literally-html 0.1.3
  * ESLint 2.4.2
  * Prettier 9.19.0

I can't seem to get Prettier to format template literals in any of my projects, despite it claiming to do so via `"embeddedLanguageFormatting": "auto"`.

**Comment by macdonst** [comment]
@RapidOwl I'm glad I could help. I really need to start running my Windows box at least 1 day a week to catch issues on Windows.

**Comment by RapidOwl** [comment]
I know I closed this, but after using it more, it turns out that changing the line endings setting wasn't the magical fix I hoped for. After searching a lot, I found [this discussion on the Lit repository](https://github.com/lit/lit/discussions/4033) and I've managed to get it working.

TL;DR Using ESLint with the Prettier and Lit plugins lets me format html tagged template literals on Windows. Then I use the lit-html extension for syntax highlighting.

Here's my example repository: https://github.com/RapidOwl/enhance-example
